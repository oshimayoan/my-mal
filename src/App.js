// @flow strict-local

import React, {Component} from 'react';
import JikanAPI from 'jikanjs';
import ImageFrame from './core-ui/ImageFrame';

import type {Season, AnimeType} from 'jikanjs';

type AnimeList = {|
  id: number,
  url: string,
  title: string,
  imageUri: string,
  type: AnimeType,
|};

type State = {|
  isLoadAnimeList: boolean,
  animeList: Array<AnimeList>,
|};

class App extends Component<{}, State> {
  state = {
    isLoadAnimeList: true,
    animeList: [],
  };

  componentDidMount() {
    this._fetchSeasonalAnime(2018, 'summer');
  }

  render() {
    let {isLoadAnimeList, animeList} = this.state;
    let animes: Array<ReactNode> = [];
    switch (isLoadAnimeList) {
      case true: {
        return <div>Fetching data...</div>;
      }
      default: {
        if (animeList.length < 1) {
          return <div>Nothing here</div>;
        }
        for (let i = 0; i < 3; i++) {
          animes = [
            ...animes,
            <ImageFrame
              key={i}
              imageUri={animeList[i].imageUri}
              title={animeList[i].title}
              link={animeList[i].url}
              dataTest={`imageFrame${i}`}
            />,
          ];
        }
        return (
          <div style={{display: 'flex', flex: 1}} data-testid="appContainer">
            {animes}
          </div>
        );
      }
    }
  }

  _fetchSeasonalAnime = async (year: number, season: Season) => {
    try {
      let result = await JikanAPI.loadSeason(year, season);
      this.setState({
        animeList: result.season.map((anime) => ({
          id: anime.mal_id,
          url: anime.url,
          title: anime.title,
          imageUri: anime.image_url,
          type: anime.type,
        })),
        isLoadAnimeList: false,
      });
    } catch (err) {
      this.setState({isLoadAnimeList: false});
    }
  };
}

export default App;
