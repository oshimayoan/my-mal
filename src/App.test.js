// @flow strict

import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('App', () => {
  it('should render without problem', () => {
    let component = renderer.create(<App />);
    expect(component).toMatchSnapshot();
  });
});
