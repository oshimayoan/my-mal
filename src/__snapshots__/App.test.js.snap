// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without problem 1`] = `
<div>
  Fetching data...
</div>
`;
